From stable-bounces@linux.kernel.org Sun Dec  2 03:19:13 2007
From: Amit Shah <amit.shah@qumranet.com>
Date: Sun,  2 Dec 2007 13:18:42 +0200
Subject: KVM: x86 emulator: Use emulator_write_emulated and not emulator_write_std
To: stable@kernel.org
Cc: kvm-devel@lists.sourceforge.net, Amit Shah <amit.shah@qumranet.com>, linux-kernel@vger.kernel.org, Avi Kivity <avi@qumranet.com>
Message-ID: <11965943272386-git-send-email-avi@qumranet.com>


From: Amit Shah <amit.shah@qumranet.com>

patch 00b2ef475d4728ca53a2bc788c7978042907e354 in mainline.

emulator_write_std() is not implemented, and calling write_emulated should
work just as well in place of write_std.

Fixes emulator failures with the push r/m instruction.

Signed-off-by: Amit Shah <amit.shah@qumranet.com>
Signed-off-by: Avi Kivity <avi@qumranet.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/kvm/x86_emulate.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/kvm/x86_emulate.c
+++ b/drivers/kvm/x86_emulate.c
@@ -1066,7 +1066,7 @@ done_prefixes:
 			}
 			register_address_increment(_regs[VCPU_REGS_RSP],
 						   -dst.bytes);
-			if ((rc = ops->write_std(
+			if ((rc = ops->write_emulated(
 				     register_address(ctxt->ss_base,
 						      _regs[VCPU_REGS_RSP]),
 				     &dst.val, dst.bytes, ctxt)) != 0)
